okapi-core/src/main/java/org/folio/okapi/util/ModuleVersionFilter.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.okapi.util;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.json.DecodeException;
+import java.util.Optional;
+import org.folio.okapi.common.ModuleId;
+
+public class ModuleVersionFilter {
+  Boolean npmSnapshot;
+  Boolean preRelease;
+
+  public ModuleVersionFilter() {
+    this(null, null);
+  }
+
+  ModuleVersionFilter(Boolean preRelease, Boolean npmSnapshot) {
+    this.preRelease = preRelease;
+    this.npmSnapshot = npmSnapshot;
+  }
+
+  /**
+   * Construct from HTTP query parameters "npmSnapshot", "preRelease".
+   * @param params HTTP parameters
+   */
+  public ModuleVersionFilter(MultiMap params) {
+    Optional<Boolean> o = getParamVersionFilter(params, "npmSnapshot");
+    npmSnapshot = o.isEmpty() ? null : o.get();
+
+    o = getParamVersionFilter(params, "preRelease");
+    preRelease = o.isEmpty() ? null : o.get();
```

Comment:
```suggestion
    npmSnapshot = getParamVersionFilter(params, "npmSnapshot").orElse(null);
    preRelease = getParamVersionFilter(params, "preRelease").orElse(null);
```
Or rewrite getParamVersionFilter to return a Boolean and not an `Optional<Boolean>`.